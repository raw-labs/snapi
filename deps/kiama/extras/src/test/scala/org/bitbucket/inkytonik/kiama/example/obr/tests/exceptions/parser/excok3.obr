(* Legal TRY statement (simple) *)

PROGRAM ExcOk3 (i : INTEGER) : INTEGER;

BEGIN

   TRY
      Average := Quantity / Count;
      Result := Average * Frequency;
   CATCH DivideByZero DO
      Result := 0;
   END
   
END ExcOk3.
