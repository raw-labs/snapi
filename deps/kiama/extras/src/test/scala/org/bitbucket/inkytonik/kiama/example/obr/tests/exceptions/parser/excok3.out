ObrInt(ExcOk3,Vector(IntParam(IdnDef(i))),Vector(TryStmt(TryBody(Vector(AssignStmt(IdnExp(IdnUse(Average)),SlashExp(IdnExp(IdnUse(Quantity)),IdnExp(IdnUse(Count)))), AssignStmt(IdnExp(IdnUse(Result)),StarExp(IdnExp(IdnUse(Average)),IdnExp(IdnUse(Frequency)))))),Vector(Catch(IdnUse(DivideByZero),Vector(AssignStmt(IdnExp(IdnUse(Result)),IntExp(0))))))),ExcOk3)
